client.go

```diff
@@ -59,19 +59,30 @@ func NewClientWithTimeout(timeout time.Duration) *Client {
 }
 
 type federationTripper struct {
-	transport http.RoundTripper
+	transports map[string]http.RoundTripper
 }
 
 func newFederationTripper() *federationTripper {
-	// TODO: Verify ceritificates
-	return &federationTripper{
-		transport: &http.Transport{
+	tripper := new(federationTripper)
+	tripper.transports = make(map[string]http.RoundTripper)
+	return tripper
+}
+
+// getTransport returns a http.Transport instance with a TLS configuration using
+// the given server name for SNI. It also creates the instance if there isn't
+// any for this server name.
+func (f *federationTripper) getTransport(tlsServerName string) http.RoundTripper {
+	if transport, ok := f.transports[tlsServerName]; !ok || transport == nil {
```

Comment:
do we need to worry about locking here?

Reply:
I guess we could for performance, otherwise if we have two concurrent calls to `getTransport` the worse thing that can happen afaik would be to replace a transport with another one doing the same thing.

Reply:
yeah, we'd also end up with two connection pools which would take a while to expire.